### PR TITLE
Implement adding "error" word to dictionary

### PR DESCRIPTION
**Pull request by VPeretiatko**

### CODE REVIEW

File: src/main/java/name/webdizz/sonar/grammar/issue/tracking/LinkFunction.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import static org.apache.commons.lang.StringUtils.stripToEmpty;
+
+public class LinkFunction implements Function, ServerExtension {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LinkFunction.class);
+
+    @Override
+    public void execute(Context context) {
+        String word = getMistakeWord(context.issue().message(), PluginParameter.ERROR_DESCRIPTION);
+        LOGGER.info("try to write:" + word);
```
Comment (webdizz) [inline src/main/java/name/webdizz/sonar/grammar/issue/tracking/LinkFunction.java]: use parameterised logging

File: src/main/java/name/webdizz/sonar/grammar/issue/tracking/LinkFunction.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+            String[] location = hostPort.split(":");
+            Preconditions.checkArgument(location.length == 2, "Illegal sonar location format in grammar settings");
+
+            HttpURLConnection connection = getConnection(word, dictName, credentials, location);
+            InputStream content = connection.getInputStream();
+            BufferedReader in = new BufferedReader(new InputStreamReader(content));
+            LOGGER.info("Saving word response:" + in.readLine());
+        } catch (IOException e) {
+            LOGGER.error("Cannot connect to Sonar for updating dictionary." + e.getMessage());
```
Comment (webdizz) [inline src/main/java/name/webdizz/sonar/grammar/issue/tracking/LinkFunction.java]: bad practice to log and throw exception, do one not both

Comment (webdizz) [inline src/main/java/name/webdizz/sonar/grammar/issue/tracking/LinkFunction.java]: loosing original exception is also a bad practice for logging and for throwing as well

File: src/main/java/name/webdizz/sonar/grammar/issue/tracking/LinkFunction.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        String dict = stripToEmpty(settings.getString(PluginParameter.MANUAL_DICT));
+        saveProperties(dict + "," + word, PluginParameter.MANUAL_DICT, context);
+    }
+
+    private String getMistakeWord(String message, String errDescription) {
+        String res = message.replaceFirst(errDescription, "");
+        return res.substring(0, res.length() - 1);
+    }
+
+    private void saveProperties(String word, String dictName, Context context) {
```
Comment (webdizz) [inline src/main/java/name/webdizz/sonar/grammar/issue/tracking/LinkFunction.java]: do not agree with this implementation - why do we need to use remote api when we're installed within Sonar? there should be an API to write properties without making a connection over http

Comment (VPeretiatko) [inline src/main/java/name/webdizz/sonar/grammar/issue/tracking/LinkFunction.java]: As far as I understand  from this thread  http://sonarqube.15.x6.nabble.com/sonar-dev-Using-settings-to-store-a-hidden-property-td5023322.html  there is no way to persist property  to db from pojo. We can read it and even write, but after restart server it disappear.

On the other hand we have "Web Service Java client" but regard  this post http://sonarqube.15.x6.nabble.com/sonar-dev-Web-Service-Java-Client-usage-tp4528518p4528523.html
we also cannot use it for persist data.


So, if you know best  way to do this  it  please inform me.

**Pull request closed without merging**